### ISSUE TITLE
using regression_coverage_score outside of MapieRegressor

### ISSUE THREAD

**Issue by remiadon** [scikit-learn-contrib/MAPIE#126]
**Is your feature request related to a problem? Please describe.**
In the context of genetic programming, I need to evaluate a given "model" (i.e a mathematical expression) on my data, many times. One advantage of GAs being the ability to use a custom fitness function, I was thinking of integrating `regression_coverage_score` inside my fitness function, so that my GA would be "looking for a performant and confident model" (still not sure of this sentence though)

From a purely practical point of view, how whould I get the `y_pred_low` and `y_pred_high` arguments to be passed to `regression_coverage_score` ? I that even possible outside of MapieRegressor ?

**Describe the solution you'd like**
```python
from mapie.metrics import regression_coverage_score
import numpy as np
from sklearn.metrics import r2_score

def fitness(model, X, y_true):
    y_pred_low, y_pred_up = function_i_am_looking_for(model, X)
    y_pred = make_prediction(model, X)  # just make a classic prediction on X
    cov = regression_coverage_score(y_true, y_pred_low, y_pred_up)
    r2 = r2_score(y_pred, y_true)
    return (r2 + cov) / 2  # take the mean but could take another weighted combination of r2 and cov
```

--------------

NB.
1. same would hold for classification_coverage_score
2. I have very limited understanding of MAPIE, please forgive my ignorance is that question ends up being silly

**Comment by vtaquet** [comment]
Hi @remiadon , thanks for your issue ! From a practical point of view, `regression_coverage_score` can be used outside of MAPIE since it simply compares `y_true` with `y_pred_low` and `y_pred_up`. All you need to do is provide `regression_coverage_score` with three arrays of same shape.

**Comment by gmartinonQM** [comment]
@remiadon, regarding the `function_i_am_looking_for` and `make_prediction`, you can try to use MAPIE indeed. MAPIE is a wrapper around a predictive model. A prective model is simply a class instance with `fit` and `predict` methods. Internally, MAPIE stores some cross-validated residuals (in regression, but analogs exist in classification), to compute the prediction interval.

**Comment by remiadon** [comment]
Hi @vtaquet and @gmartinonQM 

Thanks for the quick answer
